src/styles/calendar.scss

```diff
@@ -0,0 +1,60 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+
+  margin: $calendar-margin;
+  padding: $calendar-padding;
+  width: $calendar-width;
+  gap: $cell-gap;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    @include square($cell-size);
+    @include font-style ($font-family, $font-size);
+    cursor: pointer;
+    
+    border: 1px solid black;
+    border-radius: 5%;
+    background-color: #eee;
+
+    transition: transform 0.3s,
+      background-color 0.2s;
+
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+      box-shadow: 4px 4px teal;
+    }
+  
+    @for $index from 1 to 32 {
+      &:nth-child(#{$index}) {
+  
+        &::before {
+          content: "#{$index}";
+        }
+      }
+    }
+  }
+
+  @each $day, $graf in $start-day {
+    &__start-day_#{$day} {
+      .calendar__day:first-child {
+        margin-left: #{$graf};
+      }
+
+      display: flex;
+      flex-wrap: wrap;
+    }
+  }
+
+  @for $days from 28 to 31 {
+    &--month-length_#{$days} {
+      .calendar__day:nth-child(n + #{$days + 1}) {
+        display: none;
+      }
+    }
+  }
+}
```

Comment:
use nesting

Reply:
please explain, i thought it was nested into the proper blocks  @each is inside block "calendar" , if you are talking about selector ".calendar__day" inside my functions, so how am i supposed to nest it proper?   I need to use mixing in every "day" block? Or how?